common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -219,25 +219,56 @@ def selected_children(self):
         if hasattr(self, "_selected_set"):
             # Already done:
             return self._selected_set  # pylint: disable=access-member-before-definition
+
+        lib_tools = self.runtime.service(self, 'library_tools')
+        format_block_keys = lambda block_keys: lib_tools.create_block_analytics_summary(self.location.course_key, block_keys)
+
         # Determine which of our children we will show:
         selected = set(tuple(k) for k in self.selected)  # set of (block_type, block_id) tuples
```

Comment:
`self.selected` is the blocks that were chosen *for the current user/student*, out of the possible options in `self.children`, which in turn comes from the library.